docs/deploy/index.md

```diff
@@ -0,0 +1,148 @@
+providers such as [Amazon Web Services](https://aws.amazon.com/),
+[Google Cloud](https://cloud.google.com/), [Microsoft Azure](https://azure.microsoft.com/).
+
+- Infrastructure as a Service (IaaS): using this model you will be in charge of
+the stack from the operative system up. This is more commonly known as _virtual machines_,
+meaning that you will be in charge of choosing the python version, the web server, etc.
+At CERN we have been running Invenio v3 instances using [OpenStack](https://www.openstack.org)
+to manage the VMs, and [Puppet](https://www.puppet.com) for IaC (Infrastructure as Code).
+Other common IaC technologies are [Ansible](https://www.ansible.com) and [Terraform](https://www.terraform.io).
+
+![Deployment models](../images/deployment-models.png)
+
+You could also keep going down in the layers to get more and more control of
+your deployment, for example, using bare metal. However, that is out of the
+scope of this documentation.
+
+!!! warning "Docker compose is not recommended"
+    The provided docker-compose files can give you a good idea on what you need
+    to deploy, including the underlying services. However, the provided files
+    are **NOT** meant to be used in production as-is: it requires knowledge
+    and experience on how to run it reliable, safely stores files, understand
+    potential performance limitations, etc. Use at your own risk.
+
+## Services
+
+When deploying InvenioRDM you can choose to deploy and maintain yourself the
+services, or use third party providers. For example, the three cloud providers
+mentioned above (AWS, GC, and Azure) can provide most of the services. If you
+choose to deploy them yourself you will need to take into account the following
+topics:
+
+- Persist your data, and enable periodic backups. This includes the relational
+database, the search indices and the files.
+- Queue persistence (RabbitMQ) to avoid loosing tasks in case the service fails.
+- High availability, many of the services can be deployed redundantly. Note
+that most of the cloud providers offer this option or have an established SLA.
+- Secrets handling. Most of the services require credentials to connect. It is
+a good practice to keep them in a secure place, for example [Ansible Vault](https://www.redhat.com/sysadmin/introduction-ansible-vault).
+
+If you choose to deploy them using PaaS, images for the services can be
+found for example in [DockerHub](https://hub.docker.com). For more information
+on how to deploy each service, refer to their official documentation.
+
+## InvenioRDM Python application
+
+Independently of your deployment model choice, you will need to run the Flask
+Python application. However, the server bundled with Flask is not meant for
+production usage. Therefore, you will need some sort of WSGI HTTP server.
+Common choices are [Gunicorn](https://gunicorn.org) and [uWSGI](http://projects.unbit.it/uwsgi),
+The latter, is the one we use at CERN and it is bundled in the Dockerfile.
```

Comment:
Does it make sense to add a link to the other doc section about the RDM Docker images?